test/unit_tests/test_braket_device.py

```diff
@@ -1522,3 +1524,93 @@ def get_test_result_object(result_types=[], source="qubit[2] q; cnot q[0], q[1];
         }
     )
     return GateModelQuantumTaskResult.from_string(json_str)
+
+
+@pytest.fixture
+def noise_model():
+    return NoiseModel().add_noise(BitFlip(0.05), GateCriteria(Gate.H))
+
+
+@pytest.mark.parametrize("backend", ["braket_dm"])
+def test_valid_local_device_for_noise_model(backend, noise_model):
+    """Tests that simulator devices are analytic if ``shots`` is not supplied"""
+    dev = BraketLocalQubitDevice(wires=2, backend=backend, noise_model=noise_model)
+    assert dev._noise_model.instructions == [
+        NoiseModelInstruction(BitFlip(0.05), GateCriteria(Gate.H))
+    ]
+
+
+@pytest.mark.parametrize("backend", ["default", "braket_sv"])
+def test_invalid_local_device_for_noise_model(backend, noise_model):
+    """Tests that simulator devices are analytic if ``shots`` is not supplied"""
+    with pytest.raises(
+        ValueError,
+        match="Noise models only work on DM1 or braket_dm simulators.",
+    ):
+        _ = BraketLocalQubitDevice(wires=2, backend=backend, noise_model=noise_model)
+
+
+@pytest.mark.parametrize("device_name", ["dm1"])
+@patch.object(AwsDevice, "name", new_callable=mock.PropertyMock)
+def test_valide_aws_device_for_noise_model(name_mock, device_name, noise_model):
+    name_mock.return_value = device_name
+    dev = _aws_device(wires=2, device_type=AwsDeviceType.SIMULATOR, noise_model=noise_model)
+    assert dev._noise_model.instructions == [
+        NoiseModelInstruction(BitFlip(0.05), GateCriteria(Gate.H))
+    ]
+
+
+@pytest.mark.parametrize("device_name", ["sv1", "tn1", "foo", "bar"])
+@patch.object(AwsDevice, "name", new_callable=mock.PropertyMock)
+def test_invalide_aws_device_for_noise_model(name_mock, device_name, noise_model):
+    name_mock.return_value = device_name
+    with pytest.raises(
+        ValueError,
+        match="Noise models only work on DM1 or braket_dm simulators.",
+    ):
+        _ = _aws_device(wires=2, device_type=AwsDeviceType.SIMULATOR, noise_model=noise_model)
+
+
+@pytest.mark.parametrize("device_name", ["dm1"])
+@patch.object(AwsDevice, "run")
+@patch.object(AwsDevice, "name", new_callable=mock.PropertyMock)
+def test_execute_with_noise_model(name_mock, mock_run, device_name, noise_model):
+    mock_run.return_value = TASK
+    name_mock.return_value = device_name
+    dev = _aws_device(wires=4, device_type=AwsDeviceType.SIMULATOR, noise_model=noise_model)
+
+    with QuantumTape() as circuit:
+        qml.Hadamard(wires=0)
+        qml.QubitUnitary(1 / np.sqrt(2) * np.array([[1, 1], [1, -1]]), wires=0)
+        qml.RX(0.432, wires=0)
+        qml.CNOT(wires=[0, 1])
+        qml.probs(wires=[0])
+        qml.expval(qml.PauliX(1))
+        qml.var(qml.PauliY(2))
+        qml.sample(qml.PauliZ(3))
```

Comment:
there's already some example circuits in this file, `CIRCUIT_1`, etc. Instead of defining a new one, would it makes sense to just reuse the existing ones?

Reply:
`CIRCUIT_1` is created for adjoint gradient use case. It's not suited for testing noise model